docs/proposals/20210628-workloadspread.md

```diff
@@ -0,0 +1,223 @@
+    can be null represents that there is no replicas limits in this subset.
+- `scheduleStrategy`: strategies for schedule
+  - `type`: can be `Fixed` or `Adaptive`
+    - `Fixed` means always choose the former subset for Pods unless the subset's maxReplicas has been satisfied
+    - `Adaptive` means choose the right subset by sequence of subsets and also enough resources
+  - `adaptive`:
+    - `disableSimulationSchedule`: whether to disable the simulation schedule in webhook
+      > webhook can take a simple general predicates to check whether Pod can be scheduled into this subset, \
+        but it just considers the Node resource and cannot replace scheduler to do richer predicates practically.
+    - `rescheduleCriticalSeconds`: the critical duration that controller will trigger a Pod to reschedule to another subset.
+      > rescheduleCriticalSeconds indicates how long controller will reschedule a schedule failed Pod to the subset that has
+        redundant capacity after the subset where the Pod lives. \
+        If a Pod was scheduled failed and still in a pending status over RescheduleCriticalSeconds duration, the controller will reschedule it to the suitable subset.
+
+**status**:
+- `subsetStatuses`: statuses of subsets defined in spec
+  - `missingReplicas`: the missing number of this subset, most of the time it is (maxReplicas - the number of active Pods)
+    > missingReplicas > 0 indicates the subset is still missing MissingReplicas pods to create \
+    missingReplicas = 0 indicates the subset already has enough pods, there is no need to create \
+    missingReplicas = -1 indicates the subset's MaxReplicas not set, then there is no limit for pods number
+  - `creatingPods`: the map of Pods that just allocated to this subset by webhook
+    > creatingPods contains information about pods whose creation was processed by
+      the webhook handler but not yet been observed by the WorkloadSpread controller.
+      A pod will be in this map from the time when the webhook handler processed the
+      creation request to the time when the pod is seen by controller. \
+      The key in the map is the name of the pod and the value is the time when the webhook
+      handler process the creation request. If the real creation didn't happen and a pod is
+      still in this map, it will be removed from the list automatically by WorkloadSpread controller
+      after some time.\
+      If everything goes smooth this map should be empty for the most of the time.
+      Large number of entries in the map may indicate problems with pod creations.
+  - `deletingPods`: the map of Pods been deleted or evicted that just noticed by webhook
+    > deletingPods is similar with creatingPods and it contains information about pod deletion.
+  - `subsetUnscheduledStatus`: contains the details for the unscheduled subset.
+     - `unschedulable`: unschedulable is true indicates this Subset cannot be scheduled. The default is false.
```

Comment:
Please indicate that "unschedulable" is temporary, and the conditions for setting and restoring.